### PR TITLE
Fix service destination labels regression

### PR DESCRIPTION
**Pull request by carsonip**
## Motivation/summary

Fix regression introduced by #10056 where service destination labels are not stored successfully.

## Checklist

- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [x] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [x] Documentation has been updated

## How to test these changes

- Ensure that service destination labels are stored

## Related issues

Related to #10045

### CODE REVIEW

File: systemtest/aggregation_test.go [hunk 1]
```diff
@@ -151,13 +151,17 @@ func TestTransactionAggregationShutdown(t *testing.T) {
 }
 
 func TestServiceDestinationAggregation(t *testing.T) {
+	t.Setenv("ELASTIC_APM_GLOBAL_LABELS", "department_name=apm,organization=observability,company=elastic")
```
Comment (marclop) [inline systemtest/aggregation_test.go:154]: Can we add a numeric label so we can verify that both mappings are applied?

Comment (carsonip) [reply systemtest/aggregation_test.go:154]: Do you have an easy way to add numeric labels with the Go agent? I'm looking at Go agent tracer code and in `parseGlobalLabels` it parses labels from env into kv of string and string.

Comment (marclop) [reply systemtest/aggregation_test.go:154]: Yeah, you're right. Looking at the implementation we only ever added support for k/v string label parsing. It'd be ideal if we could test that both are parsed in APM Server, but that'd require sending a literal payload and creating another test. I think it's worth the trouble. What do you think?

Comment (carsonip) [reply systemtest/aggregation_test.go:154]: If we do it I would love to do it for all aggregations, because this test is missing for all of them.

Comment (marclop) [reply systemtest/aggregation_test.go:154]: let's open a follow up issue then to add numeric labels tests to all aggregations.

**Review by marclop — APPROVED** [review]

File: systemtest/aggregation_test.go (hunk 1 above)
Comment (carsonip) [reply systemtest/aggregation_test.go:154]: Created #10324

**Comment by marclop** [comment]
I re-verified this on the linked issue with BC5.

**Pull request merged**